packages/expo-auth-session/src/AuthRequestHooks.ts

```diff
@@ -40,23 +45,38 @@ export function useAuthRequest(
 ] {
   const [request, setRequest] = useState<AuthRequest | null>(null);
   const [result, setResult] = useState<AuthSessionResult | null>(null);
+  const isMountedRef = useRef<boolean | null>(null)
+  
+  useEffect(() => {
+    isMountedRef.current = true
+    
+    return () => {
+      isMountedRef.current = false
+    }
+  }, [])
 
   const promptAsync = useCallback(
     async (options: AuthRequestPromptOptions = {}) => {
       if (!discovery || !request) {
         throw new Error('Cannot prompt to authenticate until the request has finished loading.');
       }
       const result = await request?.promptAsync(discovery, options);
-      setResult(result);
+      isMountedRef.current && setResult(result);
       return result;
     },
     [request?.url, discovery?.authorizationEndpoint]
   );
 
   useEffect(() => {
+    let stillCareAboutThis = true
```

Comment:
```suggestion
    let isMounted = true
```

Reply:
This value does not map to the component mount state. IMO it's not correct to name it `isMounted`. That would only be try if the effect had `[]` as dependencies.

Reply:
There are still better names, even just changing the phrasing like `isHookRelevant`.

Reply:
I'm happy to change the name to whatever. I personally think `stillCareAboutThis` captures better _why_ we change the value to false on unmount then `isHookRelevant` which is very technical. `"resultOfTheAsyncOperationStilRelevant"` would be the full name I'd want to give.